### PR TITLE
add support for time and time64

### PR DESCRIPTION
**Pull request by shivanshuraj1333**
fixes https://github.com/ClickHouse/clickhouse-rs/issues/242

related to https://github.com/ClickHouse/ClickHouse/pull/81217

## Summary
A short description of the changes with a link to an open issue.

## Checklist
Delete items not relevant to your PR:
- [ ] Unit and integration tests covering the common scenarios were added
- [ ] A human-readable description of the changes was provided so that we can include it in CHANGELOG later
- [ ] For significant changes, documentation in README and https://github.com/ClickHouse/clickhouse-docs was updated with further explanations or tutorials

### CODE REVIEW

File: .idea/.gitignore [hunk 1]
```diff
@@ -0,0 +1,8 @@
+# Default ignored files
```
Comment (serprex) [inline .idea/.gitignore]: should remove all these .idea files (maybe add .idea/ to .gitignore)

Comment (shivanshuraj1333) [reply .idea/.gitignore]: sure, will do first thing in the morning

**Comment by serprex** [comment]
can remove empty `test_time_implementation.rs` file

**Comment by shivanshuraj1333** [comment]
> can remove empty test_time_implementation.rs file

Sure, was testing if things are working fine with my local clickhouse server, will clean it up.
Thanks!

File: .idea/.gitignore (hunk 1 above)
Comment (shivanshuraj1333) [reply .idea/.gitignore]: done!

**Comment by shivanshuraj1333** [comment]
working on CI failures...

File: .gitignore [hunk 2]
```diff
@@ -1,2 +1,4 @@
 target
 Cargo.lock
+
+.idea/
```
Comment (slvrtrn) [inline .gitignore]: should be removed from here, just use `.git/info/exclude` locally

File: types/src/data_types.rs [hunk 3]
```diff
@@ -66,6 +66,11 @@ pub enum DataTypeNode {
     /// Precision and optional timezone
     DateTime64(DateTimePrecision, Option<String>),
 
+    /// Optional timezone
+    Time(Option<String>),
+    /// Precision and optional timezone
+    Time64(DateTimePrecision, Option<String>),
```
Comment (slvrtrn) [inline types/src/data_types.rs:72]: But it does not have a timezone?
* https://clickhouse.com/docs/sql-reference/data-types/time
* https://clickhouse.com/docs/sql-reference/data-types/time64

File: types/src/data_types.rs [hunk 4]
```diff
@@ -379,22 +388,19 @@ fn parse_fixed_string(input: &str) -> Result<DataTypeNode, TypesError> {
...
-                err, input, size_str
+                          TypesError::TypeParsingError(format!(
+                "Invalid FixedString size, expected a valid number. Underlying error: {err}, input: {input}, size_str: {size_str}"
             ))
         })?;
         if size == 0 {
-            return Err(TypesError::TypeParsingError(format!(
-                "Invalid FixedString size, expected a positive number, got zero. Input: {}",
-                input
+                          return Err(TypesError::TypeParsingError(format!(
```
Comment (slvrtrn) [inline types/src/data_types.rs]: something is off with format here

File: types/src/data_types.rs [hunk 5]
```diff
@@ -132,10 +137,13 @@ impl DataTypeNode {
             "MultiLineString" => Ok(Self::MultiLineString),
             "Polygon" => Ok(Self::Polygon),
             "MultiPolygon" => Ok(Self::MultiPolygon),
+            "Time" => Ok(Self::Time(None)),
 
             str if str.starts_with("Decimal") => parse_decimal(str),
             str if str.starts_with("DateTime64") => parse_datetime64(str),
             str if str.starts_with("DateTime") => parse_datetime(str),
+            str if str.starts_with("Time64") => parse_time64(str),
+            str if str.starts_with("Time") => parse_time(str),
```
Comment (slvrtrn) [inline types/src/data_types.rs]: Looks like you could just return `Self::Time` here and remove `parse_time` fn, see https://clickhouse.com/docs/sql-reference/data-types/time

File: src/rowbinary/tests.rs [hunk 6]
```diff
@@ -160,3 +176,22 @@ fn it_deserializes() {
         assert_eq!(actual, sample());
     }
 }
+
+#[test]
```
Comment (slvrtrn) [inline src/rowbinary/tests.rs:180]: I'd also add a separate time/64 it for a round-trip, including corner case values

**Review by slvrtrn — CHANGES_REQUESTED** [review]

**Comment by slvrtrn** [comment]
Clippy is fixed after #254

**Comment by shivanshuraj1333** [comment]
Thanks for the review feedbacks @slvrtrn I shall update once I'm done with the changes.

File: .gitignore (hunk 2 above)
Comment (shivanshuraj1333) [reply .gitignore]: I think it's a good idea to keep it here, 3 reasons:
1) https://github.com/ClickHouse/ClickHouse/blob/master/.gitignore#L107
2) https://github.com/ClickHouse/clickhouse-rs/pull/252#discussion_r2202988885
3) I used Rustrover (form jetbrains), so if any other uses that (or vscode `.vscode`) files would be generated, can can be committed by mistake.

lmk if you want me to add `.vscode` as well.

Or if you want still want it be removed from `.gitignore`, I'm fine either ways.

File: src/rowbinary/tests.rs (hunk 6 above)
Comment (shivanshuraj1333) [reply src/rowbinary/tests.rs:180]: Makes sense, WIP

File: .gitignore (hunk 2 above)
Comment (slvrtrn) [reply .gitignore]: Let's remove it for now

Comment (shivanshuraj1333) [reply .gitignore]: Okay!

**Comment by shivanshuraj1333** [comment]
How can I trigger the CI?

**Comment by shivanshuraj1333** [comment]
tested on local, CI should pass now.

File: src/rowbinary/tests.rs (hunk 6 above)
Comment (shivanshuraj1333) [reply src/rowbinary/tests.rs:180]: Done!

File: types/src/data_types.rs (hunk 5 above)
Comment (shivanshuraj1333) [reply types/src/data_types.rs]: done

**Comment by slvrtrn** [comment]
Thanks for your contribution! Closing this in favor of https://github.com/ClickHouse/clickhouse-rs/pull/258 which is built on top of this one.
